src/main/java/org/datavyu/plugins/ffmpeg/ImageCanvasPlayerThread.java

```diff
@@ -29,25 +29,76 @@
     private static final double REFRESH_PERIOD = 0.01; // >= 1/fps
     private static final double TO_MILLIS = 1000.0;
 
+    /** x1 and y1 are respectively the x and y coordinates of the first corner of the destination rectangle. */
+    private int x1, y1;
+
+    /** x2 and y2 are respectively the x and y coordinates of the second corner of the destination rectangle. */
+    private int x2, y2;
+
     ImageCanvasPlayerThread(MediaPlayerData mediaPlayerData) {
         this.mediaPlayerData = mediaPlayerData;
         setName("Ffmpeg image canvas player thread");
         setDaemon(false);
     }
 
-    private void updateDisplay() {
-        //long time = System.nanoTime();
+    /**
+     * Draws as much of the specified area of the specified image as is
+     * currently available, scaling it on the fly to fit inside the
+     * specified area of the destination drawable surface
+     * specified by x1, y1, x2, y2 variables.
+     */
+    private synchronized void updateDisplay() {
+        Dimension size;
+        /** This loop is replacing the component listener used
```

Comment:
Can you move this to the top of the image canvas player thread...

Something like: We did not use the component listener because...